src/Exceptionless.Core/Jobs/EventPostsJob.cs

```diff
@@ -30,47 +30,77 @@ public class EventPostsJob : QueueJobBase<EventPost> {
+            }
+
+            var ep = await _storage.GetEventPostAndSetActiveAsync(queueEntry.Value.FilePath, _logger, context.CancellationToken).AnyContext();
+            if (ep == null) {
                 await AbandonEntryAsync(queueEntry).AnyContext();
                 return JobResult.FailedWithMessage($"Unable to retrieve post data '{queueEntry.Value.FilePath}'.");
             }
 
-            bool isInternalProject = eventPostInfo.ProjectId == Settings.Current.InternalProjectId;
+            bool isInternalProject = ep.ProjectId == Settings.Current.InternalProjectId;
             _logger.Info()
-                .Message("Processing post: id={0} path={1} project={2} ip={3} v={4} agent={5}", queueEntry.Id, queueEntry.Value.FilePath, eventPostInfo.ProjectId, eventPostInfo.IpAddress, eventPostInfo.ApiVersion, eventPostInfo.UserAgent)
-                .Property("ApiVersion", eventPostInfo.ApiVersion)
-                .Property("IpAddress", eventPostInfo.IpAddress)
-                .Property("Client", eventPostInfo.UserAgent)
-                .Property("Project", eventPostInfo.ProjectId)
+                .Message("Processing post: id={0} path={1} project={2} ip={3} v={4} agent={5}", queueEntry.Id, queueEntry.Value.FilePath, ep.ProjectId, ep.IpAddress, ep.ApiVersion, ep.UserAgent)
+                .Property("ApiVersion", ep.ApiVersion)
+                .Property("IpAddress", ep.IpAddress)
+                .Property("Client", ep.UserAgent)
+                .Property("Project", ep.ProjectId)
                 .Property("@stack", "event-posted")
                 .WriteIf(!isInternalProject);
 
-            var project = await _projectRepository.GetByIdAsync(eventPostInfo.ProjectId, true).AnyContext();
+            var project = await _projectRepository.GetByIdAsync(ep.ProjectId, true).AnyContext();
             if (project == null) {
-                _logger.Error().Project(eventPostInfo.ProjectId).Message("Unable to process EventPost \"{0}\": Unable to load project: {1}", queueEntry.Value.FilePath, eventPostInfo.ProjectId).WriteIf(!isInternalProject);
-                await CompleteEntryAsync(queueEntry, eventPostInfo, SystemClock.UtcNow).AnyContext();
+                _logger.Error().Project(ep.ProjectId).Message("Unable to process EventPost \"{0}\": Unable to load project: {1}", queueEntry.Value.FilePath, ep.ProjectId).WriteIf(!isInternalProject);
+                await CompleteEntryAsync(queueEntry, ep, SystemClock.UtcNow).AnyContext();
                 return JobResult.Success;
             }
 
+            byte[] uncompressedData = ep.Data;
+            if (!String.IsNullOrEmpty(ep.ContentEncoding)) {
+                try {
+                    uncompressedData = await uncompressedData.DecompressAsync(ep.ContentEncoding).AnyContext();
+                } catch (OutOfMemoryException ex) {
+                    await CompleteEntryAsync(queueEntry, ep, SystemClock.UtcNow).AnyContext();
```

Comment:
Calling `uncompressedMemoryStream.ToArray()` to get the bytes can throw an out of memory exception. IF this is the case bail!